LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -64,514 +65,541 @@ public class DefaultLoRaDataRequestHandler : ILoRaDataRequestHandler
 
         public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaRequest request, LoRaDevice loRaDevice)
         {
-            if (request is null) throw new ArgumentNullException(nameof(request));
-            if (loRaDevice is null) throw new ArgumentNullException(nameof(loRaDevice));
+            ArgumentNullException.ThrowIfNull(request, nameof(request));
+            ArgumentNullException.ThrowIfNull(loRaDevice, nameof(loRaDevice));
 
             List<Task> deferredTasks = null;
+            IAsyncDisposable deviceConnectionActivity = null;
 
-            var timeWatcher = request.GetTimeWatcher();
-
-            var loraPayload = (LoRaPayloadData)request.Payload;
-            this.d2cPayloadSizeHistogram?.Record(loraPayload.Frmpayload.Length);
-
-            var payloadFcnt = loraPayload.Fcnt;
+            try
+            {
+                ExceptionDispatchInfo edi = null;
+                LoRaDeviceRequestProcessResult result = null;
 
-            var payloadFcntAdjusted = LoRaPayloadData.InferUpper32BitsForClientFcnt(payloadFcnt, loRaDevice.FCntUp);
-            this.logger.LogDebug($"converted 16bit FCnt {payloadFcnt} to 32bit FCnt {payloadFcntAdjusted}");
+                try
+                {
+                    result = await ProcessAsync();
```

Comment:
Worth adding a comment here that this method modifies and updates `edi` and `result` since it's not obvious. Ideally we could have them as `out` arguments but those are forbidden on async methods/functions.

Reply:
This becomes obsolete with 8a82a1f3a5182e059c5e29f4cc08c7495d6eb485. The commit makes the closure obvious, I think there is no further comment needed.